### DISCUSSION TITLE
Can't create supabase client from Deno edge function

### DISCUSSION THREAD

**Discussion by pablonunn** [supabase/supabase#9790] [Questions] [answered]
Hi, I'm trying to use the supabase client into a edge function and the createClient method return an error 

`TypeError: Cannot read properties of undefined (reading 'href')
    at getParameterByName (file:///src/main.ts:2416:37)
    at new GoTrueClient (file:///src/main.ts:2896:61)
    at new SupabaseAuthClient (file:///src/main.ts:3406:9)
    at SupabaseClient._initSupabaseAuthClient (file:///src/main.ts:5196:16)
    at new SupabaseClient (file:///src/main.ts:5136:26)
    at createClient (file:///src/main.ts:5243:12)
    at Server.<anonymous> (file:///src/main.ts:5246:22)
    at Server.#respond (file:///src/main.ts:111:43)
    at Server.#serveHttp (file:///src/main.ts:132:26)`

I followed the instruction in the example bellow but found errors,  I solved some of them but finally I can't get it to work.

https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/select-from-table-with-auth-rls/index.ts

This example page have some errors

1. Line 6
` import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.0.0'`
does not work, url does not exist, I've changed it to
`import { createClient } from "https://deno.land/x/supabase/mod.ts"`
2. Line
`{ global: { headers: { Authorization: req.headers.get('Authorization')! } } }`
it returns a type error, I removed it and create the client like this
`const dbClient = createClient(
    Deno.env.get("SUPABASE_URL") ?? "",
    Deno.env.get("SUPABASE_ANON_KEY") ?? ""
  );`

In addition to what I found in the example file and which needs to be reviewed, I am unable to create the client and the TypeError remains.

**Comment by laktek** [comment, marked as answer]
Do you still have this issue? I wasn't able to reproduce it locally. The module `https://esm.sh/@supabase/supabase-js@2.0.0` is available.

**Reply by pablonunn** [reply]
Perfect, now is working,  that URL wasn't working the last days.

Thank you

**Reply by laktek** [reply]
great to hear it's working for you now.

**Reply by arifwp** [reply]
Thanks, i have same issue on this, and its solved with 
import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.0.0'

**Reply by TA2018b** [reply]
I encounter an issue of auth integration of edge function. 
Calling createClient of supabase-js causes following error and edge function doesn't launch. 
It worked normally untill a few hours ago.

`worker boot error: [ERR_MODULE_NOT_FOUND] Cannot find module 'file:///var/tmp/sb-compile-edge-runtime/node_modules/localhost/@supabase/realtime-js/2.11.8/dist/module/RealtimeClient' imported from 'file:///var/tmp/sb-compile-edge-runtime/node_modules/localhost/@supabase/realtime-js/2.11.8/dist/module/index.js'`

I call edge function from browser. My code as follows.


```
import "jsr:@supabase/functions-js/edge-runtime.d.ts";
import { createClient } from 'jsr:@supabase/supabase-js@2';
Deno.serve(async (req)=>{
  // CORS configuration required for browser access
  if (req.method === 'OPTIONS') {
    return new Response('ok', {
      headers: {
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Headers": 'authorization, x-client-info, apikey, content-type'
      }
    });
  }
  try {
    // Reject request without auth header
    const supabaseClient = createClient(Deno.env.get('SUPABASE_URL') ?? '', Deno.env.get('SUPABASE_ANON_KEY') ?? '');
    const authHeader = req.headers.get('Authorization');
    if (!authHeader) {
      return new Response(JSON.stringify({"error response here"}), {
        headers: {
          'Content-Type': 'application/json',
          'Access-Control-Allow-Origin': '*',
          "Access-Control-Allow-Headers": 'authorization, x-client-info, apikey, content-type'
        },
        status: 401
      });
    }
    // Reject request from non-authenticated session
    const token = authHeader.replace('Bearer ', '');
    const { data, error } = await supabaseClient.auth.getUser(token);
    if (error || !data.user) {
      return new Response(JSON.stringify({"error response here"}), {
        headers: {
          'Content-Type': 'application/json',
          'Access-Control-Allow-Origin': '*',
          "Access-Control-Allow-Headers": 'authorization, x-client-info, apikey, content-type'
        },
        status: 401
      });
    }
    ...
```

**Reply by firzanruzain** [reply]
I have the same issue since yesterday and now it works when i import the supabaseClient from 'https://esm.sh/@supabase/supabase-js@2.0.0' instead of 'jsr:@supabase/supabase-js@2'

**Reply by scott-w-gray** [reply]
I was having the same error in my hosted edge functions and switching from 'jsr:@supabase/supabase-js@2' to 'https://esm.sh/@supabase/supabase-js@2.0.0' fixed the original error for me but now the client object seems different.

For example I am getting errors when trying to pass in the schema to the object. 'supabaseClient.schema is not a function' and it certainly used to be one.

**Reply by GaryAustin1** [reply]
> I was having the same error in my hosted edge functions and switching from 'jsr:@supabase/supabase-js@2' to 'https://esm.sh/@supabase/supabase-js@2.0.0' fixed the original error for me but now the client object seems different.
> 
> For example I am getting errors when trying to pass in the schema to the object. 'supabaseClient.schema is not a function' and it certainly used to be one.

You can't use 2.0.0 you need to fix it to a more recent value like 2.49.8 as that version did not have .schema or many other things.
